inc/Module/Build/Sqitch.pm

```diff
@@ -224,4 +225,57 @@ sub fix_shebang_line {
     return $self->SUPER::fix_shebang_line(@_);
 }
 
+sub ACTION_bundle {
+    my ($self, @params) = @_;
+    my $base = $self->install_base or die "No --install_base specified\n";
+    $self->_runtime_cpanfile;
+    require Carton::CLI;
+    SHHH: {
+        local $SIG{__WARN__} = sub {};
+        Carton::CLI->new->cmd_install(
+            '--path'     => $base,
+            '--cpanfile' => 'cpanfile.run',
+        );
+    }
+    $self->depends_on('install');
+    $self->_add_carton_lib;
+}
+
+sub _runtime_cpanfile {
+    my $self = shift;
+    return if -e 'cpanfile.run';
+    open my $in, '<', 'cpanfile' or die "Cannot open cpanfile: $!\n";
+    open my $out, '>', 'cpanfile.run' or die "Cannot open cpanfile.run: $!\n";
+    while (<$in>) {
+        print {$out} $_ if /\Are(quire|commend)s\b/;
+    }
+    close $in;
+    close $out;
+}
+
+sub _add_carton_lib {
```

Comment:
If the goal is to produce a single distributable tree then I think this can be greatly simplfiied. Just throw this in all the scripts:

```perl
use FindBin qw( $Bin );
use lib "$Bin/../lib";
```

Maybe this could even be added to the scripts in the repo rather than munging them here.

Reply:
Yeah, `t/sqitch` does that.